Ast/src/Parser.cpp

```diff
@@ -2615,6 +2639,72 @@ AstExpr* Parser::parseString()
         return reportExprError(location, {}, "String literal contains malformed escape sequence");
 }
 
+AstExpr* Parser::parseInterpString()
+{
+    std::vector<AstArray<char>> strings;
+    std::vector<AstExpr*> expressions;
+
+    Location startLocation = lexer.current().location;
+
+    do {
+        auto currentLexeme = lexer.current();
+        LUAU_ASSERT(currentLexeme.type == Lexeme::InterpStringBegin || currentLexeme.type == Lexeme::InterpStringMid || currentLexeme.type == Lexeme::InterpStringEnd);
+
+        Location location = currentLexeme.location;
+
+        Location startOfBrace = Location(location.end, 1);
+
+        scratchData.assign(currentLexeme.data, currentLexeme.length);
+
+        if (!Lexer::fixupQuotedString(scratchData))
+        {
+            nextLexeme();
+            return reportExprError(startLocation, {}, "Interpolated string literal contains malformed escape sequence");
+        }
+
+        AstArray<char> chars = copy(scratchData);
+
+        nextLexeme();
+
+        strings.push_back(chars);
+
+        if (currentLexeme.type == Lexeme::InterpStringEnd)
+        {
+            // INTERP CODE REVIEW: I figure this isn't the right way to do this.
+            // From what I could gather, I'm expected to have strings and expressions be TempVector from the beginning.
+            // Everything that does that uses a scratch value.
+            // But I would think I would also be expected to use an existing scratch, like `scratchExpr`, in which case
+            // my assumption is that a nested expression would clash the scratches?
```

Comment:
No, TempVector is structured such that it's safe to use nested calls as long as you don't have two TempVectors in a single function that use the same scratch that you're filling concurrently.